kuttl_tests.md

```diff
@@ -0,0 +1,69 @@
+Writing KUTTL test for an operator
+==================================
+
+Kuttl is a tool for testing Kubernetes operators. Each kuttl test suite
+consists of a series of test steps that might modify the state of the cluster
+or check its state. A kuttl test step will use all yaml files that begin with
+its numeric index. For example, if we had a file 00-deploy.yaml and
+a 00-assert.yaml, kuttl would apply the changes described by the file
+00-deploy.yaml and then wait until the state of the cluster matches the one
+found in 00-assert.yaml. If it's found, then the step is successful. If after
+the timeout, the state of the cluster does not match the assert, then the test
+fails. Kuttl also supports errors files, which will check that the cluster does
+not match the state described in the file.
+
+What should be tested for in a kuttl suite depends on the details of the
+operator itself. A good place to find more information is the controller.go
+file for the operator, there you should find
+hints on what resources does the operator create and/or manage. Another source
+of information would be to contact the people listed as approvers in the
+operator's `OWNERS` file. Below are some example of things to assert while
+writing the tests (they might not apply to all operators):
+
+* Creation (the operator should deploy properly the service in question)
+* Scale up and down (the operator should control properly the number of
+replicas in the deployment and create and destroy pods as needed)
+* Destroy (the operator should delete properly the resources deployed)
+* Update (the operator applies some change to the resources deployed). As
+an example, the keystone-operator can change the configuration of the
+keystone service, which entails deleting the keystone pod and creating
+a new one with the modified configuration.
+
+Below are some conventions that are assumed for kuttl tests in
+openstack-k8s-operators:
+
+* There should be a kuttl-test.yaml file at the root of the operator
+repo which contains the kuttl configuration.
+* There should be `${OPERATOR_NAME}_kuttl` and `${OPERATOR_NAME}_kuttl_run` targets in `install_yamls`. The first will install all dependencies, run the  tests and cleanup, while the latter will simply run the tests, without further modifying the environment.
+* There should be an assert file under
+`operator-path/tests/kuttl/common/assert_sample_deployment.yaml`.
+This assert file should be used to validate the deployment
+of the operator. In parallel, a target called `${OPERATOR_NAME}\_deploy_validate`
```

Comment:
Should `${OPERATOR_NAME}\_deploy_validate` be `${OPERATOR_NAME}_deploy_validate` instead (i.e. remove the `\`)?

Reply:
Fixed, thanks!